cartoframes/context.py

```diff
@@ -249,11 +247,63 @@ def read(self, table_name, limit=None, decode_geom=False, shared_user=None):
         result = self.copy_client.copyto_stream(query)
         df = pd.read_csv(result)
 
+        return self._clean_DataFrame_from_carto(df, table_name, schema)
+
+    def _clean_DataFrame_from_carto(self, df, table_name, schema='public', decode_geom=False):
+        """
+        Clean a DataFrame with a dataset from CARTO:
+        - remove the_geom_webmercator
+        - use cartodb_id as DataFrame index
+        - process date columns
+        - decode geom
+
+        :param df: DataFrame with a dataset from CARTO
+        :type df: pandas DataFrame
+        :param table: table name in user's CARTO account.
+        :type table: str
+        :param schema: table schema name in user's CARTO account.
+        :type schema: str
+        :param decode_geom: Decodes CARTO's geometries into a `Shapely <https://github.com/Toblerity/Shapely>`__
+              object that can be used, for example, in `GeoPandas <http://geopandas.org/>`__.
+        :type decode_geom: bool, optional
+
+        :return: DataFrame
+        """
+        if 'the_geom_webmercator' in df.columns:
+            df.pop('the_geom_webmercator')
+
+        if 'cartodb_id' in df.columns:
+            df.set_index('cartodb_id', inplace=True)
```

Comment:
Just adding a comment because there's a related [issue](https://github.com/CartoDB/cartoframes/issues/580) about this, and we may change this logic in the near feature.

I'm not sure if we should expose the `cartodb_id` column to the user... Let's discuss in the issue the best approach because it involves both writing and reading dataframes.